app/lib/three_scale/spam_protection/protector.rb

```diff
@@ -56,11 +57,11 @@ def level
       end
 
       def captcha_required?
-        not logged_in? and level == :captcha
+        !logged_in? && level == :captcha && captcha_configured?
```

Comment:
```ruby
(captcha_configured && level == :captcha) && !logged_in?
```
For clarity and to not even check for `logged_in?` if captcha is not really configured